code/game/objects/effects/mines.dm

```diff
@@ -19,6 +19,7 @@
 	var/datum/weakref/foot_on_mine
 
 /obj/effect/mine/Initialize(mapload)
+	armed = TRUE
 	. = ..()
 	if(arm_delay)
 		armed = FALSE
```

Comment:
this line can be removed, then you can move `armed = TRUE` to an else on this

Reply:
i think that introduces higher chances of bugs if someone eventually edits the if chain.